packages/react-aria-components/src/Table.tsx

```diff
@@ -373,13 +373,16 @@ function Table(props: TableProps, ref: ForwardedRef<HTMLTableElement>) {
     }
   });
 
+  let isListDraggable = !dragAndDropHooks?.isDisabled && hasDragHooks;
+  let isListDroppable = !dragAndDropHooks?.isDisabled && hasDropHooks;
```

Comment:
Perhaps this should be 
```suggestion
  let isListDraggable = hasDragHooks && dragState.isDisabled;
  let isListDroppable = hasDragHooks && dropState.isDisabled;
```
That way we use the `isDisabled` set on the state objects themselves and we can have `useDragAndDrop` only return the hooks still?